spec/inputs/irc_spec.rb

```diff
@@ -1,3 +1,72 @@
 # encoding: utf-8
-require "logstash/devutils/rspec/spec_helper"
-require 'logstash/inputs/irc'
+require_relative "../spec_helper"
+require "logstash/plugin"
+require "logstash/event"
+require "cinch"
+
+describe LogStash::Inputs::Irc do
+
+  let(:properties) { {:name => "foo" } }
+  let(:event)      { LogStash::Event.new(properties) }
+
+  let(:host)       { "irc.freenode.org" }
+  let(:channels)   { ["foo", "bar"] }
+
+  it "should register without errors" do
+    plugin = LogStash::Plugin.lookup("input", "irc").new({ "host" => host, "channels" => channels })
+    expect { plugin.register }.to_not raise_error
+  end
+
+  describe "receive" do
```

Comment:
We'll I actually used the word mock in the wrong way here, the right explanation to this is that throught:

```
allow(msg).to receive(:message).and_return("message")
allow(msg).to receive(:command).and_return("command")
allow(user).to receive(:host).and_return("host")
allow(user).to receive(:nick).and_return("nick")
allow(msg).to receive(:user).and_return(user)
allow(msg).to receive(:prefix).and_return("prefix")
allow(msg).to receive(:channel).and_return("channel")
``` 

and specially:

```    let(:channel) { bot.handlers.find(:channel).first } ```

and 

```
 let(:events) do
      input(subject) do |queue|
        nevents.times do
          channel.call(msg, [], [])
        end
        result = nevents.times.inject([]) do |acc|
          acc << queue.pop
        end
        result
      end
    end
```

messages are injected to the boot as if they were actually coming from the internet. The important line here is ```          channel.call(msg, [], []) ``` the responsible to inject into the channel the message.

Hope this bring a bit of light to the question.